### ISSUE TITLE
Embedding question

### ISSUE THREAD

**Issue by chenchen-eng** [lidaof/eg-react#341]
Hello, teacher. I would like to embed Eg-react into my own webpage using your example code. How can I change the genome information and annotation details? It seems that your documentation doesn't provide instructions on this. Thank you, teacher

**Comment by lidaof** [comment]
Hi @chenchen-eng, thanks for using the Browser! In the documentation, https://eg.readthedocs.io/en/latest/installation.html#embedding, please notice the line contains `genomeName: "mm10",`, this is where you can change the genome, in the following `tracks` array you can change annotation and other tracks. Hope this helps.

**Comment by lidaof** [comment]
Correct, currently the example code only supports the genomes we loaded. In your case, you need to add your own genome following https://eg.readthedocs.io/en/latest/add.html

**Comment by lidaof** [comment]
> and install enter error:npm ERR! cb() never called!
> 
> npm ERR! This is an error with npm itself. Please report this error at: npm ERR! https://npm.community
> 
> npm ERR! A complete log of this run can be found in: npm ERR! /root/.npm/_logs/2023-11-29T14_05_52_098Z-debug.log i should how to do?thanks

please try using node version 18 or 16, we are testing on node version 20 and above.

**Comment by chenchen-eng** [comment]
thanks teacher ,but met a error
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm ERR! code ENOMEM
npm ERR! syscall spawn
npm ERR! errno -12
npm ERR! spawn ENOMEM
npm ERR! code ENOMEM
npm ERR! syscall spawn
npm ERR! errno -12
npm ERR! spawn ENOMEM

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2023-11-30T12_13_58_478Z-debug-0.log
we should how to solve it Thanks

**Comment by chenchen-eng** [comment]
use docker solve the question. but encount other question:
 ? Something is already running on port 3000. Probably:
  /usr/local/lighthouse/softwares/nodejs/node/bin/node /usr/local/lighthouse/softwares/nodejs/app/app.js (pid 1928)
  in /

Would you like to run the app on another port instead? › (Y/n)
thanks teacher

**Comment by lidaof** [comment]
Hi @chenchen-eng you can type Y here to allow the App run on another port

**Comment by chenchen-eng** [comment]
thanks teacher,please Are there any requirements for server configuration? Can my online server run with 2G of memory?

**Comment by lidaof** [comment]
> thanks teacher,please Are there any requirements for server configuration? Can my online server run with 2G of memory?

Hi @chenchen-eng if you need run `npm start` or `npm run build`, it's recommended to have at least 8G RAM
for hosting the built scripts with Apache, then any server with >512M RAM should work

**Comment by lidaof** [comment]
Hi @chenchen-eng 

try using the zip file below:
https://wangftp.wustl.edu/~dli/test/frontend.zip
if you are using node v18, type export NODE_OPTIONS=--openssl-legacy-provider  before npm start
no need to run `npm install` with this file.
